pathology/tumor_detection/torch/camelyon_train_evaluate_pytorch_gpu.py

```diff
@@ -0,0 +1,527 @@
+        preprocess_gpu_valid = Compose(
+            [
+                ToCupy(dtype=np.float32),
+                CuCIM(name="scale_intensity_range", a_min=0.0, a_max=255.0, b_min=-1.0, b_max=1.0),
+                ToTensor(device=device),
+            ]
+        )
+    elif cfg["backend"] == "numpy":
+        preprocess_cpu_train = Compose(
+            [
+                ToTensorD(keys=("image", "label")),
+                TorchVisionD(
+                    keys="image", name="ColorJitter", brightness=64.0 / 255.0, contrast=0.75, saturation=0.25, hue=0.04
+                ),
+                ToNumpyD(keys="image"),
+                RandFlipD(keys="image", prob=cfg["prob"], spatial_axis=-1),
+                RandRotate90D(keys="image", prob=cfg["prob"]),
+                CastToTypeD(keys="image", dtype=np.float32),
+                RandZoomD(keys="image", prob=cfg["prob"], min_zoom=0.9, max_zoom=1.1),
+                ScaleIntensityRangeD(keys="image", a_min=0.0, a_max=255.0, b_min=-1.0, b_max=1.0),
+                ToTensorD(keys="image"),
+            ]
+        )
+        preprocess_cpu_valid = Compose(
+            [
+                CastToTypeD(keys="image", dtype=np.float32),
+                ScaleIntensityRangeD(keys="image", a_min=0.0, a_max=255.0, b_min=-1.0, b_max=1.0),
+                ToTensorD(keys=("image", "label")),
+            ]
+        )
+    else:
+        raise ValueError(f"Backend should be either numpy or cucim! ['{cfg['backend']}' is provided.]")
+    # _________________________________________________________________________
+    # Post-processing
+    postprocess = Compose(
+        [
+            Activations(sigmoid=True),
+            AsDiscrete(threshold_values=True),
+        ]
+    )
+    # _________________________________________________________________________
+    # Create MONAI dataset
+    train_json_info_list = load_decathlon_datalist(
+        data_list_file_path=cfg["dataset_json"],
+        data_list_key="training",
+        base_dir=cfg["data_root"],
+    )
+    valid_json_info_list = load_decathlon_datalist(
+        data_list_file_path=cfg["dataset_json"],
+        data_list_key="validation",
+        base_dir=cfg["data_root"],
+    )
+    train_dataset = PatchWSIDataset(
+        data=train_json_info_list,
+        region_size=cfg["region_size"],
+        grid_shape=cfg["grid_shape"],
+        patch_size=cfg["patch_size"],
+        transform=preprocess_cpu_train,
+        image_reader_name="openslide" if cfg["use_openslide"] else "cuCIM",
+    )
+    valid_dataset = PatchWSIDataset(
+        data=valid_json_info_list,
+        region_size=cfg["region_size"],
+        grid_shape=cfg["grid_shape"],
+        patch_size=cfg["patch_size"],
+        transform=preprocess_cpu_valid,
+        image_reader_name="openslide" if cfg["use_openslide"] else "cuCIM",
+    )
+    # _________________________________________________________________________
```

Comment:
I would suggest to remove the "___________________" comments in the code.

Thansk.

Reply:
Sure, I'll do.